### PR TITLE
Add new call to check that the dev server build is complete

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
This is an additional safeguard to ensure that sideloading doesn't initiate before the dev build is completed

### CODE REVIEW

File: packages/office-addin-debugging/src/start.ts [hunk 1]
```diff
@@ -35,6 +35,15 @@ export async function isDevServerRunning(port: number): Promise<boolean> {
     return isRunning;
 }
 
+export async function isDevBuildComplete(sourceLocation: string): Promise<boolean> {
```
Comment (akrantz) [inline packages/office-addin-debugging/src/start.ts]: You could name this to reflect what it is actually doing, which is just seeing if the specified sourceLocation url can be fetched. You could make the parameter name simply "url", and call it isUrlAvailable().

File: packages/office-addin-debugging/src/start.ts [hunk 2]
```diff
@@ -100,6 +109,16 @@ export async function runDevServer(commandLine: string, port?: number): Promise<
                 } else {
                     throw new Error(`The dev server is not running on port ${port}.`);
                 }
+
+                const manifestInfo: any = await readManifestFile(manifestPath);
+                const sourceLocation: string = manifestInfo.defaultSettings.sourceLocation;
+                const isBuildComplete: boolean = await waitUntilDevBuildIsComplete(sourceLocation);
```
Comment (akrantz) [inline packages/office-addin-debugging/src/start.ts]: Per my previous comment, you can update naming so it is `const isAvailable: boolean = await waitUntilUrlIsAvailable(sourceLocation);`

File: packages/office-addin-debugging/src/start.ts [hunk 3]
```diff
@@ -100,6 +109,16 @@ export async function runDevServer(commandLine: string, port?: number): Promise<
...
+
+                if (isBuildComplete) {
+                    console.log(`The dev build completed successfully and fetch of dev source at ${sourceLocation} was successful`);
```
Comment (akrantz) [inline packages/office-addin-debugging/src/start.ts]:
```
if (isUrlAvailable) {
   console.log(`Url is available: ${sourceLocation}`);
} else {
   throw new Error(`The source location is not available from the dev server: ${sourceLocation}`);
}

File: packages/office-addin-debugging/src/start.ts [hunk 4]
```diff
@@ -214,7 +233,7 @@ export async function startDebugging(manifestPath: string, appType: AppType, app
     }
 
     if (devServerCommandLine) {
-        devServerPromise = runDevServer(devServerCommandLine, devServerPort);
+        devServerPromise = runDevServer(devServerCommandLine, manifestPath, devServerPort);
```
Comment (akrantz) [inline packages/office-addin-debugging/src/start.ts]: A new optional parameter should be added to the end, so it is not a breaking api change.

File: packages/office-addin-debugging/src/start.ts [hunk 5]
```diff
@@ -267,6 +286,10 @@ export async function waitUntil(callback: (() => Promise<boolean>), retryCount:
     return done;
 }
 
+export async function waitUntilDevBuildIsComplete(sourceLocation: string, retryCount: number = 30, retryDelay: number = 1000): Promise<boolean> {
```
Comment (akrantz) [inline packages/office-addin-debugging/src/start.ts]: Along lines of previous comments, change name to `waitUntilUrlIsAvailalbe` and change `sourceLocation` to `url`.

File: packages/office-addin-debugging/src/start.ts (hunk 1 above)
Comment (TCourtneyOwen) [reply packages/office-addin-debugging/src/start.ts]: made change locally

**Comment by akrantz** [comment]
@TCourtneyOwen What's the status of this PR?

**Comment by TCourtneyOwen** [comment]
@akrantz I think it should be good to go. I will go ahead and merge it

**Pull request merged**